openwisp_monitoring/check/classes/config_applied.py

```diff
@@ -0,0 +1,35 @@
+from swapper import load_model
+
+from ...device.utils import SHORT_RP
+from ..settings import CONFIG_CHECK_MAX_TIME
+from .base import BaseCheck
+
+AlertSettings = load_model('monitoring', 'AlertSettings')
+
+
+class ConfigApplied(BaseCheck):
+    def check(self, store=True):
+        # If the device is down do not run config applied checks
+        if self.related_object.monitoring.status == 'critical':
+            return
+        if not hasattr(self.related_object, 'config'):
+            return
+        result = 1 if self.related_object.config.status == 'applied' else 0
+        if store:
+            self.get_metric().write(
+                result, retention_policy=SHORT_RP,
+            )
+        return result
+
+    def get_metric(self):
+        metric, created = self._get_or_create_metric(configuration='config_applied')
+        if created:
+            self._create_alert_setting(metric)
+        return metric
+
+    def _create_alert_setting(self, metric):
+        alert_s = AlertSettings(
+            metric=metric, operator='<', value=1, seconds=CONFIG_CHECK_MAX_TIME * 60
```

Comment:
keep in mind this will have to be changed when we do https://github.com/openwisp/openwisp-monitoring/issues/106